### PR TITLE
Add the `TWebFile::Close(Option_t *option)` method

### PR DESCRIPTION
**Pull request by bellenot**

### CODE REVIEW

**Comment by bellenot** [comment]
@phsft-bot build just on windows10/default

**Comment by bellenot** [comment]
@phsft-bot build just on windows10/default

File: net/net/src/TWebFile.cxx [hunk 1]
```diff
@@ -405,6 +405,23 @@ Int_t TWebFile::ReOpen(Option_t *mode)
...
 }
 
+////////////////////////////////////////////////////////////////////////////////
+/// Close a Web file. Close the socket connection and delete the cache
+/// See also the TFile::Close() function
+
+void TWebFile::Close(Option_t *option)
+{
+   if (fSocket)
+      delete fSocket;
```
Comment (pcanal) [inline net/net/src/TWebFile.cxx:415]:
```suggestion
   delete fSocket;
```

File: net/net/src/TWebFile.cxx [hunk 2]
```diff
@@ -405,6 +405,23 @@ Int_t TWebFile::ReOpen(Option_t *mode)
...
+   fSocket = nullptr;
+   if (fFullCache) {
+      free(fFullCache);
+      fFullCache = 0;
```
Comment (pcanal) [inline net/net/src/TWebFile.cxx:419]:
```suggestion
      fFullCache = nullptr;
```

**Comment by bellenot** [comment]
@pcanal thanks for the review. Since it is a backport, should I also change the master and then backport those changes? And BTW, this is not my code but should I review the whole source code each time I change something nevertheless?

**Comment by pcanal** [comment]
> Since it is a backport, should I also change the master and then backport those changes?

Yes, the change technically should be in the master.   However they are NFC, so not urgent.

**Comment by bellenot** [comment]
> > Since it is a backport, should I also change the master and then backport those changes?
> 
> Yes, the change technically should be in the master. However they are NFC, so not urgent.

So I can merge this PR and apply your required changes later on in master and backport them to 6-28-00-patches, right?

**Comment by pcanal** [comment]
Yes.

**Review by pcanal — APPROVED** [review]
LGTM.  Further NFC can be done in a later PR.

**Comment by smuzaffar** [comment]
@bellenot ,  @Axel-Naumann , we are trying to test latest ROOT 6.28 [changes](https://github.com/root-project/root/compare/96ca920f34bf6ec1e8f70ed253c7e4228ea0d3cc...v6-28-00-patches) and our CI tests are failing with `Too many open files` and system logs shows [a]. 

Note that 96ca920f34bf6ec1e8f70ed253c7e4228ea0d3cc worked fine for CMSSW, it is just the last two commits on 6-28 patches branch which are causing this error. Could this PR be the reason for these failures?

[a]
```
Oct 19 15:23:09 cmsbuild120.cern.ch cvmfs2[6803]: (unpacked.cern.ch) open file descriptor limit exceeded
Oct 19 15:23:09 cmsbuild120.cern.ch cvmfs2[6803]: (unpacked.cern.ch) open file descriptor limit exceeded
Oct 19 15:23:09 cmsbuild120.cern.ch cvmfs2[6803]: (unpacked.cern.ch) open file descriptor limit exceeded
Oct 19 15:23:09 cmsbuild120.cern.ch cvmfs2[6803]: (unpacked.cern.ch) open file descriptor limit exceeded
Oct 19 15:23:09 cmsbuild120.cern.ch cvmfs2[6803]: (unpacked.cern.ch) open file descriptor limit exceeded
Oct 19 15:23:09 cmsbuild120.cern.ch cvmfs2[6803]: (unpacked.cern.ch) open file descriptor limit exceeded
Oct 19 15:23:09 cmsbuild120.cern.ch cvmfs2[6803]: (unpacked.cern.ch) open file descriptor limit exceeded
Oct 19 15:23:09 cmsbuild120.cern.ch cvmfs2[6803]: (unpacked.cern.ch) open file descriptor limit exceeded
```

**Comment by pcanal** [comment]
Not sure why yet, but indeed very likely related to the addition of `TWebFile::Close`.

**Comment by pcanal** [comment]
The patch is semantically wrong (the new code should have been after the `TFile::Close` but I also can see why it would lead to the error ... and I can not reproduce it.

**Comment by bellenot** [comment]
Sorry about that. I don't see how this change can lead to the errors you see, but anyway, how can I reproduce the error?

**Comment by smuzaffar** [comment]
I am also not sure how TWebFile can cause this. I have tried to run tests multiple times 9 on different build nodes) and every time some random unit tests fail with same error (`Too many open files`). I am now running the tests without this change to see if they still fail or not

**Comment by bellenot** [comment]
OK, thanks

**Comment by smuzaffar** [comment]
So looks like issue is not with this change. All ROOT versions 6.28 and above (i.e 6.28, 6.30 and master) open a lot of files as compare to root 6.26. See https://github.com/cms-sw/cmssw/issues/43077 issue.

**Comment by bellenot** [comment]
OK, thanks @smuzaffar

> All ROOT versions 6.28

You mean even with previous version of 6.28 (e.g. 6.28.06)?

**Comment by smuzaffar** [comment]
yes , root 6.28 patches branch even before this change. I do not know when this behavior started (I am testing it in previous builds of root 628 releases to see if I can identify it)

**Comment by bellenot** [comment]
OK, thanks @smuzaffar !

**Pull request merged**